### ISSUE TITLE
Review user input fields and the dirty data they may unintentionally but not maliciously provide

### ISSUE THREAD

**Issue by markmacgillivray** [OAButton/discussion#1570]
Add catches to user input on finds where what they send may be allowed but is clearly a copy/paste error on their end, or has characters we don't like, such as weird MS word quotes.

Also we used to have a catch for anything with the word "test" in it, because users go to the demos and put "test" in things, which was making a mess of the catalogue. But this was too rough and was blocking legitimate articles with the word "test" in them (of course). So review this too and see if it can be done more neatly.

**Comment by markmacgillivray** [comment]
Review complete, but still to write a method to catch and clean anything where appropriate.

For catching things with "test" in them, not sure this is really a good approach. Is it still necessary? Would be better to direct people to the dev site to try things out, and can connect things that are definitely demos to the dev site too, so that at least live does not get dirty data in it. But still, some users might see obvious test output coming from it. I think if the UI makes it clear they are trying a demo though, that should be OK.

**Comment by JosephMcArthur** [comment]
It is still necessary, at least for InstantILL (the only place where we allow user to overwrite data).

I've forgotten how we originally built this... so a discussion may be helpful. However, have we tried just not updating the cache with user inputted info when on dev, demos, or setup pages? I seem to have a memory of this being really hard, but maybe it isn't post(during)-refactor?

I agree 'test' is not a terribly robust way to handle things, we need something better.

**Comment by markmacgillivray** [comment]
I've made a function we can add bad characters to whenever we find them. I tried searching for some particular odd ones, but some that hit the system would never end up being saved into the system anyway, so will need to respond in future when people tell us about them. The function has been linked to the OAB endpoints where suitable, but a little bit of testing is still needed before this can go live. Maybe another hour of work this week.
